core/shape-parts.cpp

```diff
@@ -39,6 +39,32 @@ const Path& BasicBezier::generate(Point endPoint, Point transitionPoint) {
   return m_path;
 }
 
+const Path& FlatBezier::generate(Point endPoint, Point transitionPoint) {
+  MCFixed flatEndY = m_flatWidth / -2; // WHY DO I HAVE TO MAKE THIS NEGATIVE TO GET IT TO WORK???
+  if (endPoint.X < transitionPoint.X) {
+    Path ep;
+    ep.push_back(endPoint);
+    m_path = BezierPath(Point(endPoint.X, flatEndY),
+                        Point(endPoint.X, flatEndY + 
+                              (transitionPoint.Y - flatEndY) * m_endHandle),
+                        Point(transitionPoint.X -
```

Comment:
This is the Bezier control point to pull left-wards on the curve from the transition point. It should be (15 - (15 * `m_transitionHandle`), 12) no matter what, since your flat section is vertical at x = 0. That's what you have here, so that's good.